core/src/main/java/nu/yona/server/device/entities/DeviceAnonymized.java

```diff
@@ -0,0 +1,75 @@
+package nu.yona.server.device.entities;
+
+import java.time.LocalDate;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.UUID;
+
+import javax.persistence.Entity;
+import javax.persistence.ManyToOne;
+import javax.persistence.Table;
+
+import nu.yona.server.entities.EntityWithUuid;
+import nu.yona.server.subscriptions.entities.UserAnonymized;
+
+@Entity
+@Table(name = "DEVICES_ANONYMIZED")
+public abstract class DeviceAnonymized extends EntityWithUuid
+{
+	private int deviceId;
+
+	private LocalDate lastMonitoredActivityDate;
+
+	@ManyToOne
+	private UserAnonymized userAnonymized;
+
+	// Default constructor is required for JPA
+	protected DeviceAnonymized()
+	{
+		super(null);
+	}
+
+	protected DeviceAnonymized(UUID id, UserAnonymized userAnonymized, int deviceId)
+	{
+		super(id);
+		Objects.requireNonNull(userAnonymized);
```

Comment:
You can also write this as `this.userAnonymized = Objects.requireNonNull(userAnonymized)`

Reply:
Funny, I've never seen that. Thanks!